addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/ExtensionParamMiner.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import java.awt.CardLayout;
+import java.awt.Font;
+import java.io.File;
+import java.nio.file.Files;
+import javax.swing.ImageIcon;
+import javax.swing.JTextPane;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionParamMiner extends ExtensionAdaptor {
+
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionParamMiner";
+
+    // The i18n prefix, by default the package name - defined in one place to make it easier
+    // to copy and change this example
+    protected static final String PREFIX = "paramminer";
+
+    /**
+     * Relative path (from add-on package) to load add-on resources.
+     *
+     * @see Class#getResource(String)
+     */
+    private static final String RESOURCES = "resources";
+
+    private static final ImageIcon ICON =
+            new ImageIcon(ExtensionParamMiner.class.getResource(RESOURCES + "/pickaxe.png"));
+
+    private static final String EXAMPLE_FILE = "example/ExampleFile.txt";
+
+    private ZapMenuItem menuExample;
```

Comment:
IMO these could be removed and add when actually needed. Leave just the (empty) extension basically.